TestSamplePPCW.test_sample_ppc_w test is brittle

Travis complains in a random fashion about:
```
=================================== FAILURES ===================================

_______________________ TestSamplePPCW.test_sample_ppc_w _______________________


self = <pymc3.tests.test_sampling.TestSamplePPCW object at 0x7fb305447358>


    def test_sample_ppc_w(self):

        data = np.random.normal(0, 1, size=200)

        with pm.Model() as model_0:

            mu = pm.Normal('mu', mu=0, sd=1)

            y = pm.Normal('y', mu=mu, sd=1, observed=data)

            trace_0 = pm.sample()

    

        with pm.Model() as model_1:

            mu = pm.Normal('mu', mu=0, sd=1, shape=len(data))

            y = pm.Normal('y', mu=mu, sd=1, observed=data)

            trace_1 = pm.sample()

    

    

        traces = [trace_0, trace_0]

        models = [model_0, model_0]

        ppc = pm.sample_ppc_w(traces, 1000, models)

    

        assert ppc['y'].shape == (1000,)

        _, pval = stats.kstest(ppc['y'], stats.norm().cdf)

>       assert pval > 0.001

E       assert 0.00054081433365560372 > 0.001
```

Does this point to a real problem or do we just need to fix the seed?

CC @aloctavodia.

--- a/pymc3/tests/test_sampling.py
+++ b/pymc3/tests/test_sampling.py
@@ -255,8 +255,6 @@ class TestSamplePPC(object):
 class TestSamplePPCW(object):
     def test_sample_ppc_w(self):
         data0 = np.random.normal(0, 1, size=500)
-        data1 = data0 + 5
-        data2 = np.concatenate((data0, data1))
 
         with pm.Model() as model_0:
             mu = pm.Normal('mu', mu=0, sd=1)
@@ -264,25 +262,18 @@ class TestSamplePPCW(object):
             trace_0 = pm.sample()
 
         with pm.Model() as model_1:
-            mu = pm.Normal('mu', mu=5, sd=1)
-            y = pm.Normal('y', mu=mu, sd=1, observed=data1)
-            trace_1 = pm.sample()
-
-        with pm.Model() as model_2:
             mu = pm.Normal('mu', mu=0, sd=1, shape=len(data0))
             y = pm.Normal('y', mu=mu, sd=1, observed=data0)
-            trace_2 = pm.sample()
+            trace_1 = pm.sample()
+
+        traces = [trace_0, trace_0]
+        models = [model_0, model_0]
+        ppc = pm.sample_ppc_w(traces, 1000, models)
+        assert ppc['y'].shape == (1000,)
 
         traces = [trace_0, trace_1]
         models = [model_0, model_1]
-        ppc = pm.sample_ppc_w(traces, 1000, models, progressbar=False)
-        assert ppc['y'].shape == (1000,)
-        _, pval = stats.ks_2samp(ppc['y'], data2)
-        assert pval > 0.001
-
-        traces = [trace_0, trace_2]
-        models = [model_0, model_2]
-        ppc = pm.sample_ppc_w(traces, 100, models, progressbar=False)
+        ppc = pm.sample_ppc_w(traces, 100, models)
         assert ppc['y'].shape == (100, 500)
 
 

--- a/pymc3/tests/test_sampling.py
+++ b/pymc3/tests/test_sampling.py
@@ -254,30 +254,36 @@ class TestSamplePPC(object):
 
 class TestSamplePPCW(object):
     def test_sample_ppc_w(self):
-        data = np.random.normal(0, 1, size=200)
+        data0 = np.random.normal(0, 1, size=500)
+        data1 = data0 + 5
+        data2 = np.concatenate((data0, data1))
+
         with pm.Model() as model_0:
             mu = pm.Normal('mu', mu=0, sd=1)
-            y = pm.Normal('y', mu=mu, sd=1, observed=data)
+            y = pm.Normal('y', mu=mu, sd=1, observed=data0)
             trace_0 = pm.sample()
 
         with pm.Model() as model_1:
-            mu = pm.Normal('mu', mu=0, sd=1, shape=len(data))
-            y = pm.Normal('y', mu=mu, sd=1, observed=data)
+            mu = pm.Normal('mu', mu=5, sd=1)
+            y = pm.Normal('y', mu=mu, sd=1, observed=data1)
             trace_1 = pm.sample()
 
-
-        traces = [trace_0, trace_0]
-        models = [model_0, model_0]
-        ppc = pm.sample_ppc_w(traces, 1000, models)
-
-        assert ppc['y'].shape == (1000,)
-        _, pval = stats.kstest(ppc['y'], stats.norm().cdf)
-        assert pval > 0.001
+        with pm.Model() as model_2:
+            mu = pm.Normal('mu', mu=0, sd=1, shape=len(data0))
+            y = pm.Normal('y', mu=mu, sd=1, observed=data0)
+            trace_2 = pm.sample()
 
         traces = [trace_0, trace_1]
         models = [model_0, model_1]
-        ppc = pm.sample_ppc_w(traces, 50, models)
-        assert ppc['y'].shape == (50, 200)
+        ppc = pm.sample_ppc_w(traces, 1000, models, progressbar=False)
+        assert ppc['y'].shape == (1000,)
+        _, pval = stats.ks_2samp(ppc['y'], data2)
+        assert pval > 0.001
+
+        traces = [trace_0, trace_2]
+        models = [model_0, model_2]
+        ppc = pm.sample_ppc_w(traces, 100, models, progressbar=False)
+        assert ppc['y'].shape == (100, 500)
 
 
 @pytest.mark.parametrize('method', [

